tests/system/Database/Live/MySQLi/NumberNativeTest.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Database\Live\MySQLi;
+
+use CodeIgniter\Test\CIUnitTestCase;
+use CodeIgniter\Test\DatabaseTestTrait;
+use Config\Database;
+use Tests\Support\Database\Seeds\CITestSeeder;
+
+/**
+ * @group DatabaseLive
+ *
+ * @internal
+ */
+final class NumberNativeTest extends CIUnitTestCase
+{
+    use DatabaseTestTrait;
+
+    private $tests;
+    protected $refresh = true;
+    protected $seed    = CITestSeeder::class;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $config = config('Database');
+
+        $this->tests = $config->tests;
+
+        $this->tests['DBDriver'] = 'MySQLi';
+    }
+
+    public function testEnableNumberNative()
+    {
+        $this->tests['numberNative'] = true;
+
+        $db1 = Database::connect($this->tests);
+
```

Comment:
We have to add this check to every test.

```suggestion
if ($db1->DBDriver !== 'MySQLi') {
    $this->markTestSkipped('Only MySQLi can complete this test.');
}
```

Reply:
Thanks! I understand and have made the changes in the next commit.